### ISSUE TITLE
Make user setting for font/UI size

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#1775]
See: https://github.com/backdrop/backdrop-issues/issues/930#issuecomment-204763453

If the CSS for the admin theme is built well this will be really easy to implement. Since that's a goal I have anyway, this is a gimme.

This will be great for those of us that want to have smaller UI as well as a great feature for accessibility. Low vision people's gotsta edit websites too!

... mayhaps a high contrast setting too...

**Comment by mikemccaffrey** [comment]
Great idea to think more about accessibility for people with restricted vision. But perhaps it might be better to approach that from a system-wide perspective, rather than starting the discussion with one possible solution to one problem.

**Comment by wesruv** [comment]
It'd require the themes to be built using EM CSS... unless there's a clever way to set browser zoom from JS or something. Can definitely make the setting, and any theme could benefit from it, but can't promise every theme will use EM's for all font-sizes.

**Comment by biolithic** [comment]
if there was some sort of standard I would make my themes conform to it.
For example, setting a REM (this might not translate to Bartik or other
themes without extra work).

For buttons or other minor elements, I can't promise EM's (for example,
setting a button height on phones).

**Comment by wesruv** [comment]
Updated issue title and description, which will flow into the ability to have a setting for base font-size for the site, per user.

@biolithic, yep. In order to have this be a user facing setting all font sizes would have to be in EM's or REM's, and the user setting would update the font-size on the HTML tag, which would inherit throughout the document.

**Comment by wesruv** [comment]
Talked about this in the design meeting, we have a green light for changing all breakpoints to EM's, and we need to think/talk about changing all font-sizes and what the approach should be.

**Comment by wesruv** [comment]
One issue that was brought up was the lack of intuitiveness of EM's over PX in fonts. A way to address that is with the IE8/9 fallback to PX on REM's:
```
.thing {
  font-size: 18px; /* IE8 fallback to px */
  font-size: 1.125rem;
}
```

**Comment by quicksketch** [comment]
Though we're also considering dropping IE8 support: https://github.com/backdrop/backdrop-issues/issues/214#issuecomment-201253535. Though I also have reservations about em/rems and their general difficulty/misunderstanding of how they work.

**Comment by wesruv** [comment]
There are some issues with REM's in IE9 as well, so it'll be a while before we can drop that fallback completely.

**Comment by wesruv** [comment]
I should probably admit, I've been an EM fanboy for many years, and want to foist them on everything on I'm on.

Those and CSS Pseudo Elements... LOVE EM!

**Comment by dboulet** [comment]
Has there been any talk about moving to SASS? It can help make some of the math involved a lot easier.  :wink:

**Comment by biolithic** [comment]
Sass for core CSS files?  How would it work?

**Comment by wesruv** [comment]
Even though I'm a HUGE Sass fan, I don't want to add that as a requirement for core.

**Comment by biolithic** [comment]
Agreed

**Comment by dboulet** [comment]
>Even though I'm a HUGE Sass fan, I don't want to add that as a requirement for core.

That’s completely understandable, it does add complexity to development. It is doable though, WordPress core has been using SASS for years already: http://wptavern.com/wordpress-core-adopts-sass-css-preprocessor.

**Comment by wesruv** [comment]
To hammer home the EM's in `@media` queries, here's a forrealsies example.

I've cranked up my Chrome default font size to 'Very Large', did not change the viewport width, and taken screenshots of the top row of the 3-3-4 layout.

**With PX in `@media` queries:**
![image](https://cloud.githubusercontent.com/assets/5607236/14545792/e06d3538-026e-11e6-8ebf-61ef6164c082.png)


**With EM in `@media` queries**
![image](https://cloud.githubusercontent.com/assets/5607236/14545775/cd41f908-026e-11e6-8d7f-91b4ff23fa7e.png)

I much prefer the latter, layout adjusting to text size increase, like it does with modern browser's zoom implementations.

Working on a PR to update all core media queries.

**Comment by wesruv** [comment]
PR for media queries ready for review: https://github.com/backdrop/backdrop/pull/1323

**Comment by Dinornis** [comment]
Does the PR need to be updated to 1.7.x ?

Somewhat on topic:  WordPress is officially **ending support for Internet Explorer versions 8, 9, and 10**, starting with WordPress 4.8 (see https://make.wordpress.org/core/2017/04/23/target-browser-coverage/)

**Comment by quicksketch** [comment]
> Does the PR need to be updated to 1.7.x ?

Yep, and this is still applying CSS to the old layouts too, we can probably leave those as-is if we're not even allowing them to be enabled on new sites.

We've dropped support for IE8. Perhaps it's time to talk about IE9 and IE10 as well.

**Comment by jenlampton** [comment]
Adding the 'good first issue' tag since there's a PR here that needs to be rebased (and likely updated, afterward)
